ratelimiter/ratelimiter_test.go

```diff
@@ -0,0 +1,74 @@
+package ratelimiter
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRateLimit(t *testing.T) {
+	rateLimiter := New(2, map[string]time.Duration{
+		"www.google.com":   50 * time.Millisecond,
+		"www.facebook.com": 50 * time.Millisecond,
+	})
+
+	next := func() error { return nil }
+
+	google, _ := http.NewRequest("GET", "https://www.google.com", nil)
+	facebook, _ := http.NewRequest("GET", "https://www.facebook.com", nil)
+	twitter, _ := http.NewRequest("GET", "https://www.twitter.com", nil)
+
+	port := 10000
+	resp := &httptest.ResponseRecorder{}
+	test := func(succeed bool, desc string) {
```

Comment:
`succeed` sounds like some outer condition, but it's just a flag that instructs how the assertions behave. Maybe rename to `shouldSucceed`?